### ISSUE TITLE
Strong type example

### ISSUE THREAD

**Issue by wer-mathurin** [SphericalKat/dart-fuzzywuzzy#2]
@SphericalKat,

First, that's very responsive to update the library :-)  Many thanks.

If I can suggest something to update the doc to reflect better type safe code. 
Otherwise the compiler will not warn you that you may pass a list of the wrong type and the getter can throw an error if casting the wrong type.  You can omit the class type in the function, because it knows the type of the Function param.

Documentation should be for typed fonctions:

```
class TestContainer {
  final String innerVal;
  TestContainer(this.innerVal);
}

extractOne<TestContainer>(
        query: 'cowboys',
        choices: [
          'Atlanta Falcons',
          'New York Jets',
          'New York Giants',
          'Dallas Cowboys'
        ].map((e) => TestContainer(e)).toList(),
        cutoff: 10,
        getter: (x) => x.innerVal
      ).toString(); // (string Dallas Cowboys, score: 90, index: 3)
```

**Comment by SphericalKat** [comment]
Ah, good suggestion, thanks! I'll take a look into it in a bit

**Comment by SphericalKat** [comment]
Closed in #ffcac28
